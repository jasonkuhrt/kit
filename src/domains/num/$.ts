/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
