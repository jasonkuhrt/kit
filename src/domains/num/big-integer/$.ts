export * as BigInteger from './$$.ts'
