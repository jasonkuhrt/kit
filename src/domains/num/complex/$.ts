export * as Complex from './$$.js'
