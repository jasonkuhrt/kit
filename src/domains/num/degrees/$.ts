// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
