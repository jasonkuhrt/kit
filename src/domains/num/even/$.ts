// Export the namespace containing all Even operations
export * as Even from './$$.ts'
