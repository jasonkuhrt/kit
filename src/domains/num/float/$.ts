// Export the namespace containing all Float operations
export * as Float from './$$.js'
