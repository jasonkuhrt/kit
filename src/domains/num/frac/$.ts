// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
