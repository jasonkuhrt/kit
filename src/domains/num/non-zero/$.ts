// Export the namespace containing all NonZero operations
export * as NonZero from './$$.ts'
