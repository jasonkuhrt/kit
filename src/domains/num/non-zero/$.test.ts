import { Test } from '#test'
import { NonZero } from './$.js'

Test.describe('is')
  .on(NonZero.is)
  .casesAsArgs(
    // Valid cases - any non-zero number
    1,
    -1,
    0.1,
    -0.1,
    100,
    -100,
    Infinity,
    -Infinity,
    // Invalid cases
    0,
    -0,
    NaN,
    '1',
    null,
    undefined,
  )
  .test()

Test.describe('tryFrom')
  .on(NonZero.tryFrom)
  .casesAsArgs(
    // Valid cases
    1,
    -1,
    0.1,
    -0.1,
    100,
    -100,
    Infinity,
    -Infinity,
    // Invalid cases - should return null
    0,
    -0,
    NaN, // NaN is technically non-zero, returns NaN
  )
  .test()
