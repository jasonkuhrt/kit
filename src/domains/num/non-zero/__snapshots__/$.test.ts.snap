// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`is > is("1") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
"1"
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-100
╠══════════════════════════════════════════════════╣ THEN RETURNS
-100
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS
100
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;
